scala/private/phases/phase_scalafmt.bzl

```diff
@@ -0,0 +1,81 @@
+#
+# PHASE: phase scalafmt
+#
+# Outputs to format the scala files when it is explicitly specified
+#
+def phase_scalafmt(ctx, p):
+    if ctx.attr.format:
+        manifest, files = _build_format(ctx)
+        _format_runner(ctx, manifest, files)
+        _format_tester(ctx, manifest, files)
+    else:
+        ctx.actions.write(
+            output = ctx.outputs.scalafmt_runner,
+            content = "",
+            is_executable = True,
+        )
+        ctx.actions.write(
+            output = ctx.outputs.scalafmt_testrunner,
+            content = "",
+            is_executable = True,
+        )
+
+def _build_format(ctx):
+    files = []
+    runner_inputs, _, runner_manifests = ctx.resolve_command(tools = [ctx.attr._fmt])
+    manifest_content = []
+    for src in ctx.files.srcs:
+        if src.path.endswith(".scala") and src.is_source:
+            file = ctx.actions.declare_file(src.short_path)
+            files.append(file)
+            args = ctx.actions.args()
+            args.add("--config")
+            args.add(ctx.file.config.path)
+            args.add(src.path)
+            args.add(file.path)
+            args.set_param_file_format("multiline")
+            args.use_param_file("@%s", use_always = True)
```

Comment:
Small question- why `use_always = True`? Have you seen issues with Bazel's decision of when to spill over?

Reply:
Never tried `False`. Do you suggest to change?

Reply:
Yes, it will save IO for small actions (I’m guessing the majority)

Reply:
When I change it to `False`, I got this
```
ERROR: /home/borkaehw/Desktop/code/rules_scala/test/scalafmt/BUILD:15:1: ScalaFmt test/scalafmt/test/scalafmt/formatted/formatted-encoding.scala.fmt.output failed: Worker strategy cannot execute this ScalaFmt action, because the command-line arguments do not contain at least one @flagfile or --flagfile=
```
Any thoughts?

Reply:
Did you change it to False or just drop the argument (and still keep the method call obviously)? Asking because I used the args without this in a PR a couple of days ago and it worked

Reply:
> Did you change it to False or just drop the argument

I tried both, same issue.

Reply:
I want to suggest the following and LMK what you think- I suggest we merge it with the use_always and then open a separate PR just with this argument dropped. Once we have a failure there I can tag bazel people to try and help us with this pinpointed issue. WDYT?

Reply:
Sounds good.